tests/ValueParsers/TimeParserTest.php

```diff
@@ -66,372 +65,405 @@ public function validInputProvider() {
+					'+0000000000001999-00-00T00:00:00Z',
+					0, 0, 0,
+					TimeValue::PRECISION_YEAR,
+					TimeFormatter::CALENDAR_GREGORIAN
+				),
 				$noPrecOpts,
 			),
 			'+2000-00-00T00:00:00Z' => array(
-				TimeValue::newFromArray( array(
-					'time' => '+0000000000002000-00-00T00:00:00Z',
-					'timezone' => 0,
-					'before' => 0,
-					'after' => 0,
-					'precision' => TimeValue::PRECISION_YEAR,
-					'calendarmodel' => TimeFormatter::CALENDAR_GREGORIAN
-			) ),
+				new TimeValue(
+					'+0000000000002000-00-00T00:00:00Z',
+					0, 0, 0,
+					TimeValue::PRECISION_YEAR,
+					TimeFormatter::CALENDAR_GREGORIAN
+				),
 				$noPrecOpts,
 			),
 			'+2010-00-00T00:00:00Z' => array(
-				TimeValue::newFromArray( array(
-					'time' => '+0000000000002010-00-00T00:00:00Z',
-					'timezone' => 0,
-					'before' => 0,
-					'after' => 0,
-					'precision' => TimeValue::PRECISION_YEAR,
-					'calendarmodel' => TimeFormatter::CALENDAR_GREGORIAN
-			) ),
+				new TimeValue(
+					'+0000000000002010-00-00T00:00:00Z',
+					0, 0, 0,
+					TimeValue::PRECISION_YEAR,
+					TimeFormatter::CALENDAR_GREGORIAN
+				),
+				$noPrecOpts,
+			),
+
+			// Optional sign character
+			'2015-01-01T00:00:00Z' => array(
+				new TimeValue(
+					'+0000000000002015-01-01T00:00:00Z',
+					0, 0, 0,
+					TimeValue::PRECISION_DAY,
+					TimeFormatter::CALENDAR_GREGORIAN
+				),
+				$noPrecOpts,
+			),
+
+			// Optional time zone
+			'2015-01-01T00:00:00' => array(
+				new TimeValue(
+					'+0000000000002015-01-01T00:00:00Z',
+					0, 0, 0,
+					TimeValue::PRECISION_DAY,
+					TimeFormatter::CALENDAR_GREGORIAN
+				),
+				$noPrecOpts,
+			),
+
+			// Actual minus character from Unicode; roundtrip with TimeDetailsFormatter
+			"\xE2\x88\x922015-01-01T00:00:00" => array(
+				new TimeValue(
+					'-0000000000002015-01-01T00:00:00Z',
+					0, 0, 0,
+					TimeValue::PRECISION_DAY,
+					TimeFormatter::CALENDAR_GREGORIAN
+				),
+				$noPrecOpts,
+			),
+
+			// Optional second
+			'2015-01-01T00:00' => array(
+				new TimeValue(
+					'+0000000000002015-01-01T00:00:00Z',
+					0, 0, 0,
+					TimeValue::PRECISION_DAY,
+					TimeFormatter::CALENDAR_GREGORIAN
+				),
+				$noPrecOpts,
+			),
+
+			// Optional hour and minute
+			'2015-01-01' => array(
+				new TimeValue(
+					'+0000000000002015-01-01T00:00:00Z',
+					0, 0, 0,
+					TimeValue::PRECISION_DAY,
+					TimeFormatter::CALENDAR_GREGORIAN
+				),
+				$noPrecOpts,
+			),
+
+			// Years < 59 require at least hour and minute
+			'1-01-01T00:00' => array(
+				new TimeValue(
+					'+0000000000000001-01-01T00:00:00Z',
+					0, 0, 0,
+					TimeValue::PRECISION_DAY,
+					TimeFormatter::CALENDAR_GREGORIAN
+				),
+				$noPrecOpts,
+			),
+
+			// Day zero
+			'2015-01-00' => array(
+				new TimeValue(
+					'+0000000000002015-01-00T00:00:00Z',
+					0, 0, 0,
+					TimeValue::PRECISION_MONTH,
+					TimeFormatter::CALENDAR_GREGORIAN
+				),
+				$noPrecOpts,
+			),
+
+			// Month zero
+			'2015-00-00' => array(
```

Comment:
Erm, sorry? What is "2015<DASH><DASH>"? This doesn't even remotely resembles an ISO string. How is that clearer?